specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2172 @@
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/registryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/buildDefinitionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly."
+          },
+          "202": {
+            "description": "The request was successful; the operation will complete asynchronously."
+          },
+          "204": {
+            "description": "No Content - the specified resource was not found."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "BuildDefinitions_Delete": {
+            "$ref": "./examples/BuildDefinitionsDelete.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "BuildDefinitions"
+        ],
+        "summary": "Updates a build definition with the specified parameters.",
+        "operationId": "BuildDefinitions_Update",
+        "parameters": [
+          {
+            "name": "buildDefinitionUpdateParameters",
+            "in": "body",
+            "description": "The parameters for updating a build definition.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BuildDefinitionUpdateParameters"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/registryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/buildDefinitionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/BuildDefinition"
+            }
+          },
+          "201": {
+            "description": "The request was successful; the operation will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/BuildDefinition"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "BuildDefinitions_Update": {
+            "$ref": "./examples/BuildDefinitionsUpdate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/buildDefinitions/{buildDefinitionName}/queue": {
```

Comment:
Why isn't this modeled as a PUT /build/{name}? This would allow both creation of the build and also add it to the queue. I agree POST /queue and POST /dequeue might make sense but that should be in addition to PUT which allows create and queue . But I dont see dequeue also.

Reply:
Build doesn't have any name as such. The Ids are generated by RP and returned in response to queue build. There is no dequeue for a  build. A build can be cancelled and it is a POST on builds